python/pyiceberg/io/pyarrow.py

```diff
@@ -151,7 +160,7 @@ def open(self) -> InputStream:
                 an AWS error code 15
         """
         try:
-            input_file = self._filesystem.open_input_file(self._path)
+            input_file = self._filesystem.open_input_stream(self._path, buffer_size=self._buffer_size)
```

Comment:
I think that we need to have a flag to signal when `seek` will not be called. The protocol includes seek and we need to support it for Parquet files. However, we can hint to Arrow that we don't need to seek. How about adding `seekable=True` to `open` and allowing the caller to use this by setting `seekable=False`?

Reply:
That's a great point. Since it is part of the protocol, we should be able to use `seek`. I've added the `seekable` option to the `.open` operator.